### PR TITLE
Update index.rst

### PR DESCRIPTION
**Pull request by fcuantico**
NWChemEx extension conventions for various file types

**Is this pull request associated with an issue(s)?**
No this is adding new content to the documentation

**Description**
Description of the extension to different file types

**TODOs**
For draft pull requests please include a list of what needs to be done and check
off items as you complete them.

### CODE REVIEW

File: docs/source/conventions/index.rst [hunk 1]
```diff
@@ -30,3 +30,9 @@ Repository Conventions
     :maxdepth: 2
 
     file_structure
+
+In NWChemEx the following convetions are followed:
+C++ header: file_name.hpp
+C++ source files: file_name.cpp
+CMake source files : filename.cmake 
+CMake Lists: CMakeLists.txt
```
Comment (jwaldrop107) [inline docs/source/conventions/index.rst:38]:
```suggestion
In NWChemEx the following convetions are followed:

- C++ header: file_name.hpp
- C++ source files: file_name.cpp
- CMake source files : filename.cmake 
- CMake Lists: CMakeLists.txt
```

The formatting needs to be corrected to be a proper list. But I would move the information intended here to the files about the particular languages, e.g. move the note on C++ file extensions to [`cxx_conventions.rst`](https://github.com/NWChemEx/.github/blob/cc42c68e89c14c76112210ae467351d78fd8374a/docs/source/conventions/cxx_conventions.rst). Or the full section can be moved to [file_structure.rst](https://github.com/NWChemEx/.github/blob/cc42c68e89c14c76112210ae467351d78fd8374a/docs/source/conventions/file_structure.rst), maybe.

**Comment by ryanmrichard** [comment]
@jwaldrop107 and @fcuantico what do you think of these changes?

**Review by jwaldrop107 — APPROVED** [review]
LGTM

**Comment by jwaldrop107** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/NWChemEx/.github/actions/runs/12548412640) **New version**:[v0.3.62](https://github.com/NWChemEx/.github/releases/tag/v0.3.62) **Changes**:[v0.3.61...v0.3.62](https://github.com/NWChemEx/.github/compare/v0.3.61...v0.3.62)

**Pull request merged**